modules/dev2plus/hw_01/test/BinaryTree.test.js

```diff
@@ -0,0 +1,127 @@
+const BinaryNode = require("../src/Node/BinaryNode");
+const { Tree } = require("../src/Tree/Tree");
+const dataSet = require("../src/assets/dataSet");
+const { BinaryTree } = require("../src/Tree/BinaryTree");
+
+const mockDataSet = {
+  value: {
+    name: "James Smith",
+    salary: 15,
+  },
+  leftChild: {
+    value: {
+      name: "John Johnson",
+      salary: 10,
+    },
+    leftChild: {
+      value: {
+        name: "Mary Williams",
+        salary: 7,
+      },
+    },
+  },
+  rightChild: {
+    value: {
+      name: "Richard Davis",
+      salary: 13,
+    },
+  },
+};
+
+test("BinaryTree obj should be an extension of the Tree", () => {
+  const BST = new BinaryTree();
+  expect(BST instanceof BinaryTree).toBe(true);
+});
+
+test("BinaryTree should have an insert method", () => {
```

Comment:
this is testing implementation details, generally not a good idea. 
also you need to call `insert` in other test cases so that will cover this